### PR TITLE
Bug 1358661 - Add tab spinner etl job

### PR DESCRIPTION
**Pull request by fbertsch**
Original ETL notebook here: https://gist.github.com/fbertsch/f08998b87174d73c6f939b911e2cce53

I may not be using this library correctly. Feel free to request significant changes :)

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/python_mozetl/15)

### CODE REVIEW

File: mozetl/tab_spinner/utils.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+def windows_only(p):
+    return p["environment/system/os/name"] == "Windows_NT"
+
+def e10s_enabled_only(p):
+    return p["environment/settings/e10sEnabled"]
+
+def long_spinners_keyed_by_build_and_client(ping):
+    return ((ping["application/buildId"][:8], ping["clientId"]), (ping["payload/histograms/FX_TAB_SWITCH_SPINNER_VISIBLE_LONG_MS"], ping["payload/histograms/FX_TAB_SWITCH_SPINNER_VISIBLE_MS"]))
+
+def none_is_subsumed(x, y):
```
Comment (harterrt) [inline mozetl/tab_spinner/utils.py]: Please document this functions intent, I'm not sure what it's supposed to do.

File: mozetl/tab_spinner/utils.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    elif hist[hist.index >= 0].sum() > 0:
+        severity[named_index[2]] = 1
+
+    return (buildId, severity)
+
+def to_percentages(build_severities):
+    severities = build_severities[1]
+    total_clients = severities.sum()
+    if total_clients > 0:
+        return (build_severities[0], severities / total_clients)
```
Comment (harterrt) [inline mozetl/tab_spinner/utils.py]: Are severities and total_counts ever both integers (causing integer division)?

Comment (fbertsch) [inline mozetl/tab_spinner/utils.py]: No, severities is a Pandas series, which seems to be fine with integer division.

**Comment by fbertsch** [comment]
@harterrt RFAL

**Comment by harterrt** [comment]
Review status: 0 of 6 files reviewed at latest revision, 5 unresolved discussions.

---

*[mozetl/tab_spinner/generate_counts.py, line 37 at r1](https://reviewable.io:443/reviews/mozilla/python_mozetl/15#-Kj3_FdBXXaqAOGYyOHv:-Kj3_FdBXXaqAOGYyOHw:b-d6xaf0) ([raw file](https://github.com/mozilla/python_mozetl/blob/eee866653703a3959aa354459120afa5ce3093c6/mozetl/tab_spinner/generate_counts.py#L37)):*
> ```Python
> 
>         pings = old_infra_pings.union(new_infra_pings)
>         build_results[build_type] = get_short_and_long_spinners(pings)
> ```

Flake8 is complaining that `get_short_and_long_spinners` is not defined. Can you verify that this function will run?

---

*[mozetl/tab_spinner/utils.py, line 12 at r1](https://reviewable.io:443/reviews/mozilla/python_mozetl/15#-Kj3ZjtXWHXiHSKWDZlB:-Kj3ZjtXWHXiHSKWDZlC:bljtgos) ([raw file](https://github.com/mozilla/python_mozetl/blob/eee866653703a3959aa354459120afa5ce3093c6/mozetl/tab_spinner/utils.py#L12)):*
> ```Python
> 
> def long_spinners_keyed_by_build_and_client(ping):
>     return ((ping["application/buildId"][:8], ping["clientId"]), (ping["payload/histograms/FX_TAB_SWITCH_SPINNER_VISIBLE_LONG_MS"], ping["payload/histograms/FX_TAB_SWITCH_SPINNER_VISIBLE_MS"]))
> ```

Please shorten lines to 79 characters, if possible here and throughout.

---

*[mozetl/tab_spinner/utils.py, line 128 at r1](https://reviewable.io:443/reviews/mozilla/python_mozetl/15#-Kj3bCxwTk-vxj6S5b5b:-Kj3bCxxHixdA6XdvKj7:b9rdtm7) ([raw file](https://github.com/mozilla/python_mozetl/blob/eee866653703a3959aa354459120afa5ce3093c6/mozetl/tab_spinner/utils.py#L128)):*
> ```Python
>     windows_pings_only = ping_props.filter(windows_only)
>     e10s_enabled_on_windows_pings_only = windows_pings_only.filter(e10s_enabled_only)
>     grouped_spinners = e10s_enabled_on_windows_pings_only.repartition(200).map(long_spinners_keyed_by_build_and_client).reduceByKey(add_tuple_series)
> ```

Please split this line.

---

*[scheduling/tab_spinner.py, line 3 at r1](https://reviewable.io:443/reviews/mozilla/python_mozetl/15#-Kj3bLdkqLSqNyKrxNYO:-Kj3bLdkqLSqNyKrxNYP:bagbeg0) ([raw file](https://github.com/mozilla/python_mozetl/blob/eee866653703a3959aa354459120afa5ce3093c6/scheduling/tab_spinner.py#L3)):*
> ```Python
> import os
> from pyspark import SparkConf, SparkContext
> from pyspark.sql import SQLContext
> ```

This import is unused.

---

*[tests/test_tab_spinner_aggregates.py, line 25 at r1](https://reviewable.io:443/reviews/mozilla/python_mozetl/15#-Kj3fiHBN7_iu0uwb_Jz:-Kj3fiHBN7_iu0uwb_K-:b-g02uo7) ([raw file](https://github.com/mozilla/python_mozetl/blob/eee866653703a3959aa354459120afa5ce3093c6/tests/test_tab_spinner_aggregates.py#L25)):*
> ```Python
> 
>     expected = {
>         "short": {"20170101": pd.Series([0, 1.0, 0, 0, 0, 0, 0, 0])},
> ```

These tests look good, but I'm not sure I understand the intended analysis. The goal to create a bucketed histogram for the `sum` field for each of these two histograms, correct? 

If so, how will this analysis be used? Is client_id vs ping weighting important?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/python_mozetl/15)*

**Comment by harterrt** [comment]
Thanks, Frank. Some formatting nits and a question to help me understand the purpose of the analysis. Let me know if you think there's someone 

---

Review status: 0 of 6 files reviewed at latest revision, 5 unresolved discussions.

---

*[tests/test_tab_spinner_aggregates.py, line 25 at r1](https://reviewable.io:443/reviews/mozilla/python_mozetl/15#-Kj3fiHBN7_iu0uwb_Jz:-Kj3j7ZotYW7fs6elljy:bgd07v8) ([raw file](https://github.com/mozilla/python_mozetl/blob/eee866653703a3959aa354459120afa5ce3093c6/tests/test_tab_spinner_aggregates.py#L25)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary><blockquote>

These tests look good, but I'm not sure I understand the intended analysis. The goal to create a bucketed histogram for the `sum` field for each of these two histograms, correct? 

If so, how will this analysis be used? Is client_id vs ping weighting important?
</blockquote></details>

Actually, scratch that. After discussing offline, our main goal here is to avoid the ATMO flakiness our clients have been hitting recently. This analysis powers a dashboard meant to identify regressions in Nightly, so the interpretability isn't super important for now either. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/python_mozetl/15#-:-Kj3hkgRKTlJFkNMwbKO:b-a4z0va)*

**Comment by fbertsch** [comment]
@harterrt fixed the missing imports and flake8 issues.

**Comment by harterrt** [comment]
Reviewed 3 of 6 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/python_mozetl/15)*

**Comment by harterrt** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/python_mozetl/15#-:-KjAPR7BxTX59Vgeigvq:bnfp4nl)*

**Pull request merged**